### ISSUE TITLE
Auto-imports for generated schema types

### ISSUE THREAD

**Issue by murshex** [johannschopplich/nuxt-api-party#82]
### Describe the feature

Hi, is it possible to have auto imports for my typescript schema? Right now I have to do this:
```javascript
import { LoginPlatform } from "#nuxt-api-party/api";
```
Would much prefer auto imports. Is that possible? Thanks

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Can you think of other implementations of this feature?

### Final checks

- [X] Check existing [issues](https://github.com/johannschopplich/nuxt-api-party/issues).

**Comment by mattmess1221** [comment]
You can configure openapi-typescript to generate schema type exports.

```ts
export default defineNuxtConfig({
  apiParty: {
    openAPITS: {
      rootTypes: true,
      rootTypesNoSchemaPrefix: true, // defaults to false, which prefixes Schema to every type
    }
  }
})
```

The module doesn't know what components are defined, so it won't be able to generate auto-imports. You'll have to define them yourself.

```ts
export default defineNuxtConfig({
  imports: {
    presets: [{
      from: '#nuxt-api-party/api',
      type: true
      imports: ['Component1', 'Component2', 'ComponentN']
    }]
  }
})
```

If you keep your schema in a json file, you could load it in a hook and read the schema names yourself.

```ts
export default defineNuxtConfig({
  hooks: {
    'imports:sources': async (presets) => {
      presets.push({
        from: '#nuxt-api-party/api',
        type: true,
        imports: Object.keys((await import('./schemas/api.json')).components.schemas)
      })
    }
  }
})
```

**Comment by johannschopplich** [comment]
Yeah, Auto-Imports should be handled per application. I don't want to enable that globally.

By the way, I just published a new set of [OpenAPI type helpers](https://nuxt-api-party.byjohann.dev/guide/openapi-types.html#openapi-type-helpers) which might come in handy for you!
